### DISCUSSION TITLE
feature:  add neotest adpter for cpp such as gtest

### DISCUSSION THREAD

**Discussion by meicale** [LazyVim/LazyVim#4123] [Extra Requests]
### Did you check the docs?

- [X] I have read all the LazyVim docs
- [X] This is not a request for a new or existing extra (see above)

### Is your feature request related to a problem? Please describe.

Using extra python, we can debug and test python code. But for c&cpp, there is a extra cofnig clangd, but it doesn't support test function.  

### Describe the solution you'd like

Add neotest adapter for gtest. This link is just for reference. https://github.com/alfaix/neotest-gtest/issues/6

### Describe alternatives you've considered

It should be good to add it. Thank you!

**Comment by dpetka2001** [comment]
PRs are welcome.

**Reply by meicale** [reply]
I tried  to config it but meet some promlems. If I can be informed how to deal with it, I will provide the pr!
